### ISSUE TITLE
renders a list instead a grid on iOS

### ISSUE THREAD

**Issue by repthor** [NobsterTheLobster/Xamarin.Forms.GridView#1]
Hi NobsterTheLobster,

Thank You very much for this! Great work! I have one issue: under iOS the same code renders as a vertical List instead a grid as it should. I use no grouping. May be, I made a mistake. I do every thing in XAML:

               <ContentView     x:Name="indexcolor_block"
                                HorizontalOptions="FillAndExpand"
                                VerticalOptions="FillAndExpand"
                                                                >
                    <ContentView.Resources>
                        <ResourceDictionary>
                            <vm:ViewModel x:Key="viewModel" />

                        </ResourceDictionary>
                    </ContentView.Resources>

                    <Frame  x:Name="frame" HorizontalOptions="Center" VerticalOptions="End" Padding="0">
                        <Frame.BackgroundColor>Red</Frame.BackgroundColor>
                        <Frame.Content>
                            <widget:GridView
			                                                    x:Name="gridview_indexcolors"
                                                                Grid.Row="1" 
                                                                IsGroupingEnabled="false"
                                                                BindingContext="{StaticResource viewModel}" 
                                                                ItemsSource="{Binding Colours}"
                                                                RowHeight="86"    
			                                                    MinItemWidth="90"
                                                                SelectedItem="{Binding SelectedItem,Mode=TwoWay}"
                                                                TappedCommand="{Binding TappedCommand}"
                                                                BackgroundColor="#202020">
                                <widget:GridView.ItemTemplate>
                                    <DataTemplate>
                                        <widget:GridViewXamlCell>
                                            <Grid Padding="5">
                                                <Grid BackgroundColor="{Binding Color}" Padding="0" Margin="0">
                                                    <Label Text="{Binding GroupId}" FontSize="Micro" TextColor="White" VerticalOptions="EndAndExpand" 
                                               HorizontalOptions="Fill" BackgroundColor="#80202020" Margin="0" HeightRequest="35" VerticalTextAlignment="Center"/>
                                                </Grid>
                                            </Grid>
                                        </widget:GridViewXamlCell>
                                    </DataTemplate>
                                </widget:GridView.ItemTemplate>
                            </widget:GridView>
                        </Frame.Content>
                    </Frame>

                </ContentView>

**Comment by repthor** [comment]
android:

![image](https://cloud.githubusercontent.com/assets/11502827/24501091/609ee8f4-1548-11e7-98ab-bb4df8403692.png)

**Comment by repthor** [comment]
Sorry, Im unable to get a screenshot from my Mac...;) But there are the same colors, but only one at one row each.

**Comment by NobsterTheLobster** [comment]
Silly question but are you sure you've added the renderer to ios. I can only think that it can't find the renderer and since the control inherits from the xamarin listview its failing over to the default xamarin listviewrenderer. 

Does the sample project work for you.

![xamaringridview_ios](https://cloud.githubusercontent.com/assets/26567276/24508200/ab40782a-155a-11e7-9741-d8944196b105.png)

**Comment by repthor** [comment]
Thaks for Your answer. I will first set a breakpoint into Your iOS renderer to check wether it is involved in rendering at all.

Then i will load your example project.

 

Thanks.

 

Thomas

 

Gesendet: Donnerstag, 30. März 2017 um 15:40 Uhr
Von: NobsterTheLobster <notifications@github.com>
An: "NobsterTheLobster/Xamarin.Forms.GridView" <Xamarin.Forms.GridView@noreply.github.com>
Cc: repthor <thomas.repmann@web.de>, Author <author@noreply.github.com>
Betreff: Re: [NobsterTheLobster/Xamarin.Forms.GridView] renders a list instead a grid on iOS (#1)


Silly question but are you sure you've added the renderer to ios. I can only think that it can't find the renderer and since the control inherits from the xamarin listview is failing over to the default xamarin listviewrenderer.

Does the sample project work for you.

**Comment by repthor** [comment]
Sorry Lobster,

 

i did some copy and paste (namespace etc.) and killed:

[assembly: ExportRenderer(typeof(GridView), typeof(GridViewRenderer))]
 

Could not work, of course. Now it is rendered fine on iOS too.

 

Thomas

 

Gesendet: Donnerstag, 30. März 2017 um 15:40 Uhr
Von: NobsterTheLobster <notifications@github.com>
An: "NobsterTheLobster/Xamarin.Forms.GridView" <Xamarin.Forms.GridView@noreply.github.com>
Cc: repthor <thomas.repmann@web.de>, Author <author@noreply.github.com>
Betreff: Re: [NobsterTheLobster/Xamarin.Forms.GridView] renders a list instead a grid on iOS (#1)


Silly question but are you sure you've added the renderer to ios. I can only think that it can't find the renderer and since the control inherits from the xamarin listview is failing over to the default xamarin listviewrenderer.

Does the sample project work for you.

**Comment by repthor** [comment]
I included a bug. 

Was no issue at all.

Sorry.

**Comment by NobsterTheLobster** [comment]
No bother at all. Actually while investigating what could be going wrong for you I realised that the group headers are not actually working for ios.. They are working in my live production app where I'm using this control but not for the sample in this repo. I figured out why and will have an update tomorrow. Wouldn't have realised there was a problem if you hadn't opened an issue so it worked out.

EDIT: By the way, really nice looking app you have there going by the screenshot.
